### ISSUE TITLE
feat: add search collator

### ISSUE THREAD

**Issue by drodil** [v-ngu/backstage-plugin-bulletin-board#11]
To make most of this great plugin, you could also provide a Search Collator for the backend to be able to search bulletins from the Backstage search. It's quite easy to do by just providing a `CollatorFactory` that implements the `DocumentCollatorFactory` from `@backstage/plugin-search-common`. You can check my Q&A plugin as a reference https://github.com/drodil/backstage-plugin-qeta/blob/main/plugins/qeta-backend/src/search/collators/QetaCollatorFactory.ts

**Comment by v-ngu** [comment]
Good idea! It would make the plugin a lot easier to use. If you want to tackle any of the suggested features, feel free to do it. Otherwise I will definitely look into it.

**Comment by drodil** [comment]
Great, I will take a look if I have time at some point to contribute!
